src/test/bufferlist.cc

```diff
@@ -2369,6 +2369,22 @@ TEST(BufferList, splice) {
     EXPECT_EQ((unsigned)11, bl.length());
     EXPECT_EQ(0, ::memcmp("fghi", other.c_str(), other.length()));
   }
+
+  {
+    /* bug replicator, to test rebuild_aligned_size_and_memory() in this
+     * special scenarios */
+    bufferlist bl1;
+    bl.clear();
+    std::vector<int> data(30);
+    bufferptr ptr(buffer::create_aligned(16384, 4096));
+
+    encode(data, bl);
+    bl.append_zero(4096 - bl.length());
+    bl.append(ptr);
+    bl.splice(0, 8192, &bl1);
+    bl.rebuild_aligned_size_and_memory(4096, 4096, 1024);
+  }
```

Comment:
Oops, looks an `EXPECT_...` clause is still missed. Not staged when amending the commit?

Reply:
> Oops, looks an `EXPECT_...` clause is still missed. Not staged when amending the commit?

sorry, not familiar with this process. Now `EXPECT_...` has been added. Please check is it ok?

Reply:
I think this should be moved to `TEST(BufferList, rebuild_aligned_size_and_memory)`.  It doesn't belong in `TEST(BufferList, splice)` because `splice()` is not the problem / what is being tested here.

Reply:
I also wonder if it might be better to avoid `splice()` in the reproducer and construct the bufferlist matching the conditions (two buffers, first is memory and size aligned, second is 0-length) manually?  That way this test case would still work even if the implementation of `splice()` changes in the future.